rustls-libssl/src/x509.rs

```diff
@@ -100,6 +143,37 @@ impl OwnedX509 {
         }
     }
 
+    /// Create a new one, from a (donated) existing ref.
+    pub fn new(raw: *mut X509) -> Self {
+        Self { raw }
+    }
+
+    /// Create a new one, by incrementing an existing ref.
+    pub fn new_incref(raw: *mut X509) -> Self {
+        unsafe { X509_up_ref(raw) };
+        Self { raw }
+    }
```

Comment:
in `EvpKey::new_incref` there's a `debug_assert!` that the `raw` arg isn't null. Worth doing the same here?